vulture_os/toolkit/api_parser/cortex_xdr/cortex_xdr.py

```diff
@@ -216,10 +222,21 @@ def format_log(self, log, kind):
     def execute(self):
         for kind in self.event_kinds.keys():
             logger.info(f"[{__parser__}]:execute: Getting {kind}", extra={'frontend': str(self.frontend)})
+            # Prevent requesting too newer alerts, sometimes case_id is empty and added later
+            # Delay of 2 minutes seems acceptable
+            to = timezone.now()-timedelta(minutes=2)
+            since = getattr(self, f"cortex_xdr_{kind}_timestamp") or timezone.now()-timedelta(hours=1)
+            if since > to:
+                logger.info(f"[{__parser__}]:execute: Last timestamp too newer, waiting next execution.",
+                            extra={'frontend': str(self.frontend)})
+                continue
+
             cpt = 0
             total = 1
             while cpt < total and not self.evt_stop.is_set():
-                reply_obj = self.get_logs(kind, nb_from=cpt, since=getattr(self, f"cortex_xdr_{kind}_timestamp"))
+                reply_obj = self.get_logs(kind, nb_from=cpt, since=getattr(self, f"cortex_xdr_{kind}_timestamp") or
+                                                                   timezone.now()-timedelta(hours=1),
+                                          to=to)
```

Comment:
You cannot change the filters of the search AND the offset count to get from the API!
the `since` value should not change during all the iterations of pages queried